### DISCUSSION TITLE
Periodic Boundary Conditions in TensorMechanics

### DISCUSSION THREAD

**Discussion by Aayush-T** [idaholab/moose#22441] [Q&A General] [answered]
Hi everyone, 
greetings!
I am trying to use periodic boundary conditions for my Tensor Mechanics model. I use mesh file of either .inp or .msh type. I am attaching a few sample mesh files and input files with a sample example. I have developed my model based on FiniteStrainUObasedCP on very similar lines as described in the example. 

On using automatic PBCs, moose throws error: *** ERROR ***
Couldn't auto-detect a paired boundary for use with periodic boundary conditions

whereas when I try using translation PBC method, it doesn't produce any error but still won't utilize the periodic boundary conditions and model will work as if PBCs are not applied. Even writing wrong/non-existant primary/secondary boundary names doesn't throw any other and the model works as if no PBCs are applied.
I verified that the model is not using PBCs by looking at disp_x from the generated .e file using paraview.

I am uploading files for both inp and msh mesh type as well as input files for both. Had to add.txt in the end as github doesn't allow a few file formats to be uploaded. Apologies for that.
[euler_ang_file.txt](https://github.com/idaholab/moose/files/9820086/euler_ang_file.txt)
[filename_updated_new.inp.txt](https://github.com/idaholab/moose/files/9820233/filename_updated_new.inp.txt)
[final.msh.txt](https://github.com/idaholab/moose/files/9820234/final.msh.txt)
[input_slip_sys.txt](https://github.com/idaholab/moose/files/9820235/input_slip_sys.txt)
[save_euler_pbc_gmsh.i.txt](https://github.com/idaholab/moose/files/9820236/save_euler_pbc_gmsh.i.txt)
[save_euler_pbc_inp.i.txt](https://github.com/idaholab/moose/files/9820237/save_euler_pbc_inp.i.txt)

**Comment by GiudGiud** [comment, marked as answer]
Hello

I just run your simulation for 3 time steps
I added
construct_side_list_from_node_list = true to help visualize x0 x1 y0 y1

the three points I checked had the expected periodicity. 
Could you please double check?
points on x0 and matching points and x1 should have the same disp_y right?

Guillaume

**Reply by Aayush-T** [reply]
Yes, on adding "construct_side_list_from_node_list = true", we get expected periodicity in both the cases, but without it, we do not get periodicity.
Thanks a lot!

**Reply by GiudGiud** [reply]
ok thanks for reporting it. I created #22496 to keep track of this

**Reply by Aayush-T** [reply]
Thank you again! Also, I wanted to know if there is some specific reason for this to happen? Or is it perfectly normal thing.

**Reply by GiudGiud** [reply]
I m not an expert on this subsystem so I dont know.
It could be a bug or it could have been intentional
